src/infi/systray/traybar.py

```diff
@@ -161,15 +191,19 @@ def _restart(self, hwnd, msg, wparam, lparam):
         self._refresh_icon()
 
     def _destroy(self, hwnd, msg, wparam, lparam):
+        self._terminate()
+
+    def _terminate(self):
+        # TODO * release self._menu with DestroyMenu and reset the member
+        #      * release self._hicon with DestoryIcon and reset the member
+        #      * release loaded menu icons (loaded in _load_menu_icon) with DeleteObject
+        #        (we don't keep those objects anywhere now)
         if self._on_quit:
             self._on_quit(self)
         nid = NotifyData(self._hwnd, 0)
         Shell_NotifyIcon(NIM_DELETE, ctypes.byref(nid))
         PostQuitMessage(0)  # Terminate the app.
-        # TODO * release self._menu with DestroyMenu and reset the memeber
-        #      * release self._hicon with DestoryIcon and reset the member
-        #      * release loaded menu icons (loaded in _load_menu_icon) with DeleteObject
-        #        (we don't keep those objects anywhere now)
+        DestroyWindow(self._hwnd)

```

Comment:
So, just to clarify, both self._destroy() and SysTrayIcon.QUIT were OK before and we don't need self._terminate() at all, right?